docker/entrypoint.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+echo "Waiting for database..."
+while ! python -c "import socket; s = socket.socket(); s.connect(('db', 5432))"; do
+  sleep 1
+done
```

Comment:
Use a more robust command to check the database readiness, such as `pg_isready`, which is specifically designed for PostgreSQL:
```
while ! pg_isready -h db -p 5432 -U "$DB_USER"; do
  echo "Waiting for PostgreSQL to be ready..."
  sleep 1
done
```

Alternatively, if you prefer Python, handle exceptions to avoid potential script termination:
  ```
  while ! python -c "import socket; s = socket.socket(); s.connect(('db', 5432))" 2>/dev/null; do
    echo "Waiting for PostgreSQL to be ready..."
    sleep 1
  done
```